core/src/main/scala/org/apache/spark/memory/DefaultMemoryManager.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.memory
+
+import java.nio.ByteBuffer
+
+import com.gemstone.gemfire.internal.snappy.UMMMemoryTracker
+import com.gemstone.gemfire.internal.snappy.memory.MemoryManagerStats
+import io.snappydata.collection.ObjectLongHashMap
+
+import org.apache.spark.{Logging, SparkEnv}
+import org.apache.spark.storage.BlockId
+
+
+class DefaultMemoryManager extends StoreUnifiedManager with Logging {
+
+  private val memoryForObject: ObjectLongHashMap[(String, MemoryMode)]
+  = ObjectLongHashMap.withExpectedSize[(String, MemoryMode)](16)
+
+
+  private val managerId = "DefaultMemoryManager"
+
+  override def acquireStorageMemoryForObject(objectName: String,
+      blockId: BlockId,
+      numBytes: Long,
+      memoryMode: MemoryMode,
+      buffer: UMMMemoryTracker,
+      shouldEvict: Boolean): Boolean = synchronized {
+    logDebug(s"Acquiring DefaultManager memory for $objectName $numBytes")
+    val env = SparkEnv.get
+    if (env ne null) {
+      val success = env.memoryManager.acquireStorageMemory(blockId, numBytes, memoryMode)
+      memoryForObject.addTo(objectName -> memoryMode, numBytes)
+      success
+    } else {
```

Comment:
Wouldn't it be better to do a sync on env.memoryManager because its call already uses the same?

```scala
  override def acquireStorageMemoryForObject(objectName: String,
      blockId: BlockId,
      numBytes: Long,
      memoryMode: MemoryMode,
      buffer: UMMMemoryTracker,
      shouldEvict: Boolean): Boolean = {
    logDebug(s"Acquiring DefaultManager memory for $objectName $numBytes")
    val env = SparkEnv.get
    if (env ne null) {
     env.memoryManager.synchronized {
       val success = env.memoryManager.acquireStorageMemory(blockId, numBytes, memoryMode)
       memoryForObject.addTo(objectName -> memoryMode, numBytes)
       success
    }
   } else {
```